src/__tests__/dialog.spec.tsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react'
+import '@testing-library/jest-dom'
+import { requestAPI } from '../utils/util';
+import { when } from 'jest-when'
+
+jest.mock("../utils/util", () => ({
+    requestAPI: jest.fn()
+}))
+
+const DEFAULT_VALID_API_KEY = "valid_api_key"
+const DEFAULT_INVALID_API_KEY = "invalid_api_key"
+const DEFAULT_PROJECT_ID = "project_id_abc"
+const DEFAULT_JOB_ID = "job_id_abc"
+
+
+const mockEnvWithAPI = (APIKey: any = null) => {
+    when((requestAPI as jest.Mock)).calledWith("apikey").mockResolvedValue({
+        data: APIKey
+    })
+}
+
+const mockPostAPI = (result: any, api_key: string) => {
+    when((requestAPI as jest.Mock)).calledWith("apikey", { "body": `{\"api_key\":\"${api_key}\"}`, "method": "POST" }).mockResolvedValue({
+        result: result
+    })
+}
+
+// If projectID is provided, the API is used for re-deployment
+const mockJobDeployResult = (projectID = "", deployment_result: object = {
+    project_id: DEFAULT_PROJECT_ID,
+    id: DEFAULT_JOB_ID
+}) => {
+    when((requestAPI as jest.Mock)).calledWith("job", { "body": `{\"api_key\":\"${DEFAULT_VALID_API_KEY}\",\"project_id\":\"${projectID}\"}`, "method": "POST" }).mockResolvedValue({
+        deployment_result: deployment_result
+    })
+}
+
+const renderDialogContent = (project_id: any = null) => {
+    if (!project_id) {
+        render(<DialogContent metadata={new Map()} />);
+    } else {
+        render(<DialogContent metadata={new Map([["ploomber", { project_id: project_id }]])} />);
+    }
+}
+
+describe("Test DialogContent ", () => {
+
+
+    describe("API Key Input Stage", () => {
+
+        /* Before:
+            1. No API Key saved in user's environment
+        */
+        beforeEach(() => {
+            jest.clearAllMocks();
+            mockEnvWithAPI()
+        });
+
+        /*
+        Ensure layout components render properly
+        */
+        test("Test API Key Layout", async () => {
+            renderDialogContent();
+            await waitFor(() => {
+                expect(screen.getByLabelText('API Key')).toBeVisible()
+                expect(screen.getByRole('link')).toHaveTextContent('Click here to get an API Key')
+                expect(screen.getByRole('button')).toHaveTextContent('CONFIRM')
+            })
+        })
+
+        /* 
+        Test Flow: When the user inputs invalid API Key
+        */
+        test("Test Invalid API Key Input", async () => {
+            mockPostAPI("fail", DEFAULT_INVALID_API_KEY)
+            renderDialogContent();
+            await waitFor(() => {
+                const inputBox = screen.getByLabelText('API Key')
+                fireEvent.change(inputBox, { target: { value: DEFAULT_INVALID_API_KEY } })
+                const confirmButton = screen.getByText('CONFIRM', { selector: 'button' })
+                fireEvent.click(confirmButton)
+            })
+            await waitFor(() => {
+                expect(screen.getByLabelText('Please enter valid API Key')).toBeVisible()
```

Comment:
ok so here you're testing what happens when the API key is invalid but I don't understand why one value triggers the invalid key when both of the sample values provided are invalid API keys (DEFAULT_VALID_API_KEY and DEFAULT_INVALID_API_KEY)

Reply:
I guess I just want to demonstrate there will be `VALID_API_KEY` and `INVALID_API_KEY` use cases in the test cases, 

It's more about showing the concept, do you think its better to put the actual format API_KEY like `v5k8Y_lqy8DKDeVCKI97kyalUL_bINUZvg_0qKNIGVw`?

Reply:
so I guess you're mocking the responses right? because the API key is not a real key. that's fine for now, I think in an upcoming iteration we can add a valid key but let's leave it as is for now